src/csharp/Microsoft.Spark.Extensions.Delta.UnitTest/DeltaTableTests.cs

```diff
@@ -0,0 +1,180 @@
+using Microsoft.Spark.Sql;
+using Xunit;
+
+namespace Microsoft.Spark.Extensions.Delta.UnitTest
+{
+    /// <summary>
+    /// These tests will fail if the Delta Jar is not installed in the 
+    /// $SPARK_HOME/jars/ directory.
+    /// </summary>
+    [Collection(Constants.DeltaTestContainerName)]
+    public class DeltaTableTests
+    {
+        private readonly SparkSession _spark;
+
+        public DeltaTableTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+        }
+
+        /// <summary>
+        /// Run the end-to-end scenario from the Delta Quickstart tutorial.
+        /// </summary>
+        /// <see cref="https://docs.delta.io/latest/quick-start.html"/>
+        [Fact]
+        public void TestTutorialScenario()
+        {
+            using TemporaryDirectory tempDirectory = new TemporaryDirectory();
+            string path = Path.Combine(tempDirectory.Path, "delta-table");
+
+            // Write data to a Delta table.
+            DataFrame data = _spark.Range(0, 5);
+            data.Write().Format("delta").Save(path);
+
+            // Validate that data contains the the sequence [0 ... 4].
+            ValidateTutorialDataFrame(Enumerable.Range(0, 5), data);
+
+            // Create a second iteration of the table.
+            data = _spark.Range(5, 10);
+            data.Write().Format("delta").Mode("overwrite").Save(path);
+
+            // Load the data into a DeltaTable object.
+            var deltaTable = DeltaTable.ForPath(path);
+
+            // Validate that deltaTable contains the the sequence [5 ... 9].
+            ValidateTutorialDataFrame(Enumerable.Range(5, 5), deltaTable.ToDF());
+
+            //// Update every even value by adding 100 to it.
+            deltaTable.Update(
+                condition: Functions.Expr("id % 2 == 0"),
+                set: new Dictionary<string, Column>()
+                {
+                    { "id", Functions.Expr("id + 100") }
+                });
+
+            //// Validate that deltaTable contains the the data:
+            //// +---+
+            //// | id|
+            //// +---+
+            //// |  5|
+            //// |  7|
+            //// |  9|
+            //// |106|
+            //// |108|
+            //// +---+
+            ValidateTutorialDataFrame(new List<int>() { 5, 7, 9, 106, 108 }, deltaTable.ToDF());
+
+            // Delete every even value.
+            deltaTable.Delete(condition: Functions.Expr("id % 2 == 0"));
+
+            // Validate that deltaTable contains:
+            // +---+
+            // | id|
+            // +---+
+            // |  5|
+            // |  7|
+            // |  9|
+            // +---+
+            ValidateTutorialDataFrame(new List<int>() { 5, 7, 9 }, deltaTable.ToDF());
+
+            // Upsert (merge) new data.
+            DataFrame newData = _spark.Range(0, 20).As("newData").ToDF();
+
+            deltaTable.As("oldData")
+                .Merge(newData, "oldData.id = newData.id")
```

Comment:
line 133 uses == but only a single = is used here ?

Reply:
The Delta tutorial seems to use both interchangeably.

Reply:
ok sounds good.